### PR TITLE
Filtering for the leaderboard + UI changes for app name/version

### PR DESCRIPTION
**Pull request by sfc-gh-jreini**
# Description

- Add leaderboard filters for app_name and app_version, tags and app metadata. 
- Display tags in info-button.
- Display app_name/app_version where app_id was displayed
- Populate app_name/app_version/app_metadata in evaluations table
- Sort by app_name/app_version

Demo:

https://github.com/user-attachments/assets/9782a39e-8954-40e8-94a6-94702a725cba

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] New Tests
- [x] This change includes re-generated golden test results
- [ ] This change requires a documentation update

### CODE REVIEW

File: examples/dev/dummy_app/filter_apps.ipynb [hunk 1]
```diff
@@ -0,0 +1,72 @@
+{
```
Comment (sfc-gh-dkurokawa) [inline examples/dev/dummy_app/filter_apps.ipynb:1]: This notebook doesn't show anything about filtering. Did you mean to show some other notebook?

File: examples/dev/dummy_app/filter_apps.ipynb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    "sys.path.append(str(repo_path))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from trulens.core import Tru\n",
```
Comment (sfc-gh-dkurokawa) [inline examples/dev/dummy_app/filter_apps.ipynb:24]: Should use `TruSession` now.

File: examples/dev/dummy_app/filter_apps.ipynb (hunk 1 above)
Comment (sfc-gh-jreini) [reply examples/dev/dummy_app/filter_apps.ipynb:1]: updated - this PR is just UI side changes, but the notebook is to give you easy apps with metadata to play with.

File: src/dashboard/trulens/dashboard/pages/Evaluations.py [hunk 3]
```diff
@@ -149,24 +149,44 @@ def extract_metadata(row: pd.Series) -> str:
     return str(record_data["meta"])
 
 
-apps = list(app["app_id"] for app in lms.get_apps())
+app_ids = list(app["app_id"] for app in lms.get_apps())
+
+
+@st.cache_data
```
Comment (sfc-gh-chu) [inline src/dashboard/trulens/dashboard/pages/Evaluations.py:155]: if we cache apps, new apps won't be reflected in the dashboard without manually reloading

File: src/dashboard/trulens/dashboard/Leaderboard.py [hunk 4]
```diff
@@ -30,12 +30,104 @@ def leaderboard():
...
 
-    # Set the title and subtitle of the app
+    # Set the title
     st.title("App Leaderboard")
-    st.write(
-        "Average feedback values displayed in the range from 0 (worst) to 1 (best)."
+
+    # wrapper so we can cache the records and feedback
+    @st.cache_data
+    def get_data():
```
Comment (sfc-gh-chu) [inline src/dashboard/trulens/dashboard/Leaderboard.py:38]: same here

File: src/dashboard/trulens/dashboard/Leaderboard.py [hunk 5]
```diff
@@ -30,12 +30,104 @@ def leaderboard():
...
+        return lms.get_records_and_feedback([])
+
+    @st.cache_data
+    def get_apps():
```
Comment (sfc-gh-chu) [inline src/dashboard/trulens/dashboard/Leaderboard.py:42]: and here

File: src/dashboard/trulens/dashboard/pages/Evaluations.py (hunk 3 above)
Comment (sfc-gh-jreini) [reply src/dashboard/trulens/dashboard/pages/Evaluations.py:155]: addressed all

**Comment by sfc-gh-jreini** [comment]
Note: this PR will get much smaller when https://github.com/truera/trulens/pull/1392 is merged to main

**Review by sfc-gh-pdharmana — APPROVED** [review]

**Pull request merged**